File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    let maxW = 0;
    let maxH = 0;
    var answer = 0;
    sizes.forEach(([w,h])=>{
        // 둘 중 하나가 maxSize 보다 크다면,
        if(w < h){
            const temp = h;
            h = w;
            w = temp;
        }
        maxW = Math.max(maxW, w);
        maxH = Math.max(maxH, h);
        
    });
    
    return maxW * maxH;
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    max_w = 0
    max_h = 0
    for size in sizes :
        if size[0] < size[1] :
            temp = size[1]
            size[1] = size[0]
            size[0] = temp
        max_w = max(max_w, size[0])
        max_h = max(max_h, size[1])
    return max_w*max_h